public/apps/configuration/panels/user-list.tsx

```diff
@@ -103,12 +106,17 @@ export function UserList(props: AppDependencies) {
   const [currentUsername, setCurrentUsername] = useState('');
   const [loading, setLoading] = useState(false);
   const [query, setQuery] = useState<Query | null>(null);
+  const { dataSource, setDataSource } = useContext(DataSourceContext)!;
 
   React.useEffect(() => {
     const fetchData = async () => {
       try {
         setLoading(true);
-        const userDataPromise = getUserList(props.coreStart.http, ResourceType.users);
+        const userDataPromise = getUserList(
+          props.coreStart.http,
+          ResourceType.users,
+          createDataSourceQuery(dataSource.id)
+        );
         setCurrentUsername((await getAuthInfo(props.coreStart.http)).user_name);
```

Comment:
This is used to disable the delete button on the current user - does this check still make sense/should we keep it such that you cannot delete a same-named user in the remote cluster? I just kept it there for now since it seems sensible.